src/lowered.jl

```diff
@@ -122,7 +122,7 @@ end
 function methods_by_execution(mod::Module, ex::Expr; kwargs...)
     methodinfo = MethodInfo()
     docexprs = DocExprs()
-    value, frame = methods_by_execution!(JuliaInterpreter.Compiled(), methodinfo, docexprs, mod, ex; kwargs...)
+    value, frame = methods_by_execution!(Base.inferencebarrier(JuliaInterpreter.Compiled()), methodinfo, docexprs, mod, ex; kwargs...)
```

Comment:
This seems to be an actual usecase example of https://github.com/JuliaLang/julia/pull/41931. As far as I understand, does it turn out to be effective to turn off inference of `recurse(...)` within a callstack of `methods_by_execution!`? If `recurse(...)` is less-frequently called actually, then that sounds a reasonable hack to improve latency.

Reply:
Yes, this is exactly a `@noinfer`. All the execution time specific to a particular `recurse` is in JuliaInterpreter anyway.